containers/src/Kokkos_OffsetView.hpp

```diff
@@ -925,6 +925,151 @@ class OffsetView : public ViewTraits<DataType, Properties...> {
     Mapping::assign(m_map, rhs.m_map, rhs.m_track);  // swb what about assign?
   }
 
+ private:
+  enum class subtraction_failure {
+    none,
+    non_negative,
+    overflow,
+  };
+
+  // Subtraction should return a strictly positive number and not overflow
+  template <typename I>
+  KOKKOS_INLINE_FUNCTION static subtraction_failure check_subtraction(I lhs,
+                                                                      I rhs) {
+    if (lhs <= rhs) return subtraction_failure::non_negative;
+
+    if (std::is_signed<I>::value) {
+      using U = typename std::make_unsigned<I>::type;
+      U diff  = static_cast<U>(lhs) - static_cast<U>(rhs);
+      U max   = (static_cast<U>(0) - static_cast<U>(1)) >> 1;
+      if (diff > max) return subtraction_failure::overflow;
```

Comment:
I am having a hard time to follow.  Someone else review this.

Reply:
Why do we need this? Whats wrong with if(begin[i] > end[I]) Also I don't get how you are legally casting lhs or rhs to an unsigned? I mean isn't one of the primary use cases of OffsetView to have negative begins? That said I haven't fully thought this through I guess. I.e. what happens if I use -1.5B to +1B or something like that as the bounds. Also in general we have not been checking carefully for overflow risks. The costs of that are just too high. Like the access operator of View is templated on the integer types provided, since we don't want to cast in between. But that means if you use 

```c++
int i,j,k;
...
a(i,j,k);
```
to access the whole thing you better be sure that this doesn't overflow in the index calculation. I guess we should add a check during debug that it doesn't though.

Reply:
Negative begins/ends is fine (although I realize I no longer have tests for that).  The unsigned check is merely looking for overflowing the subtraction, and should be fairly cheap given that I'm already testing for begins[I] < ends[I].